kong/clustering/init.lua

```diff
@@ -216,6 +216,45 @@ function _M:request_version_negotiation()
   end
 end
 
+local function call_control_plane(premature, self)
+  if premature then
+    return
+  end
+
+  local wsjs_dp = require "kong.clustering.data_plane"
+  local wrpc_dp = require "kong.clustering.wrpc_data_plane"
+
+  ngx_log(ngx_DEBUG, _log_prefix, "will try wrpc first")
+
+  local implementation = wrpc_dp
+  local ws_conn, err = implementation.open_connection(self)
+  ngx_log(ngx_DEBUG, _log_prefix, "got conn: ", tostring(ws_conn), ", err: ", tostring(err))
+
+  -- TODO: find how to detect a 404 response
```

Comment:
?

Reply:
the websocket library doesn't return the http result code, instead, returns the whole header line if it doesn't match the expected pattern.  this, added to some inherited fallbacks in the specific `server` context in nginx config, makes reliably detecting a 404 response a bit more involved than expected.  i left the specifics to after i get the negotiation itself working.  (that is, now). 

the `"got conn:"` log messages are just to see the exact returned values. they won't stay in the final code

Reply:
as it turns out, any HTTP result is "good enough" to the library, so the user never sees a 404 or any other error.

I've just hacked a peeking socket to work around it.